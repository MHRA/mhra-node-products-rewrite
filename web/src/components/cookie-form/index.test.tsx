import renderer from 'react-test-renderer';
import React from 'react';
import CookieForm from './index';

describe(CookieForm, () => {
  it('should render 🍪', () => {
    // tslint:disable-next-line: no-empty
    const noop = () => {};
    const component = renderer
      .create(<CookieForm storageAllowed setStorageAllowed={noop} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should render 🚫', () => {
    // tslint:disable-next-line: no-empty
    const noop = () => {};
    const component = renderer
      .create(<CookieForm storageAllowed={false} setStorageAllowed={noop} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
