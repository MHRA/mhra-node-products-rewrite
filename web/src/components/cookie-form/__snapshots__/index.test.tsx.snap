// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookieForm should render 🍪 1`] = `
<section
  className="sc-bczRLJ iahJwb"
>
  <h3
    className="sc-dkzDqf kQvSSC"
  >
    Toggle cookie permissions
  </h3>
  <form
    onSubmit={[Function]}
  >
    <p>
      <label
        htmlFor="cookie-on"
      >
        <input
          aria-checked={true}
          checked={true}
          id="cookie-on"
          name="cookie"
          onChange={[Function]}
          type="radio"
          value="on"
        />
        On
      </label>
      <label
        htmlFor="cookie-off"
      >
        <input
          aria-checked={true}
          checked={false}
          id="cookie-off"
          name="cookie"
          onChange={[Function]}
          type="radio"
          value="off"
        />
        Off
      </label>
    </p>
    <div
      className="sc-gsnTZi jZbapK"
      dangerouslySetInnerHTML={
        Object {
          "__html": Object {
            "process": [Function],
          },
        }
      }
    />
    <p>
      <button>
        Save your preferences
      </button>
    </p>
  </form>
</section>
`;

exports[`CookieForm should render 🚫 1`] = `
<section
  className="sc-bczRLJ iahJwb"
>
  <h3
    className="sc-dkzDqf kQvSSC"
  >
    Toggle cookie permissions
  </h3>
  <form
    onSubmit={[Function]}
  >
    <p>
      <label
        htmlFor="cookie-on"
      >
        <input
          aria-checked={false}
          checked={false}
          id="cookie-on"
          name="cookie"
          onChange={[Function]}
          type="radio"
          value="on"
        />
        On
      </label>
      <label
        htmlFor="cookie-off"
      >
        <input
          aria-checked={false}
          checked={true}
          id="cookie-off"
          name="cookie"
          onChange={[Function]}
          type="radio"
          value="off"
        />
        Off
      </label>
    </p>
    <div
      className="sc-gsnTZi jZbapK"
      dangerouslySetInnerHTML={
        Object {
          "__html": Object {
            "process": [Function],
          },
        }
      }
    />
    <p>
      <button>
        Save your preferences
      </button>
    </p>
  </form>
</section>
`;
